### PR TITLE
A11y: Fix fastpass issues for dashboard management pages

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Fixes fastpass issues for dashboard management pages as part of the ongoing fastpass fixes.
See #39429.

### CODE REVIEW

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍 

one small question: i didn't get the addition of the ids on various elements - is that a fastpass requirement (or just a general a11y thing) that inputs have ids? 🤔

File: public/app/features/search/components/SearchResultsFilter.tsx [hunk 1]
```diff
@@ -42,7 +42,11 @@ export const SearchResultsFilter: FC<Props> = ({
 
   return (
     <div className={styles.wrapper}>
-      {editable && <Checkbox value={allChecked} onChange={onToggleAllChecked} />}
+      {editable && (
+        <div className={styles.checkboxWrapper}>
+          <Checkbox label="Select all" value={allChecked} onChange={onToggleAllChecked} />
```
Comment (Clarity-89) [inline public/app/features/search/components/SearchResultsFilter.tsx:47]: Would `aria-label` be enough here maybe? That label looks a bit out of place there imo.

**Review by Clarity-89 — APPROVED** [review]
Looks good!

**Comment by Clarity-89** [comment]
> one small question: i didn't get the addition of the ids on various elements - is that a fastpass requirement (or just a general a11y thing) that inputs have ids? 🤔

The way our `Field` component currently works is if its child element has an `id` prop, the Field will [automatically add](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Forms/Field.tsx#L79-L88) `htmlFor` attribute to the label for that element.

**Comment by ashharrison90** [comment]
ahhh thanks, I forgot it did it automatically, I thought you had to provide pairs of ids like normal html

**Review by mckn — APPROVED** [review]
LGTM!

**Pull request merged**